test/integration/scheduler/preemption_test.go

```diff
@@ -900,6 +898,11 @@ func TestNominatedNodeCleanUp(t *testing.T) {
 	if err := waitForNominatedNodeName(cs, medPriPod); err != nil {
 		t.Errorf("NominatedNodeName annotation was not set for pod %v/%v: %v", medPriPod.Namespace, medPriPod.Name, err)
 	}
+	// And medium priority pod should be unschedulable.
+	if err := confirmPodIsUnschedulable(cs, medPriPod); err != nil {
+		t.Errorf("Pod %v/%v was not unschedulable: %v", medPriPod.Namespace, medPriPod.Name, err)
+	}
+
```

Comment:
nit: I'd say this is not _that_ necessary, as in https://github.com/kubernetes/kubernetes/pull/90978#issuecomment-627153537, I've explained the med and high Pods would never be abled to "scheduled" in this test - as low Pods have non-zero GracePeriod set, and no kubelet runtime can confirm the deletion of those low Pods.

However, if you do want to add this check, `confirmPodIsUnschedulable` can be flaky - it only fetch the resource once, if there is a network glitch, the returned error would fail the whole test. So if with the check, `confirmPodIsUnschedulable` should be tweaked a bit to be more fault-tolerant.

Reply:
OK.  I realized it would never really be scheduled, due to the grace period and resource limitations, but the idea behind this check was just to make sure the pod condition status reason got set to Unschedulable as expected. 

I don't think there is any integration test to check this currently and there is code in the scheduler that sets the pod condition to unschedulable.

I would be glad to remove this if you think it is unnecessary.

Do you still think I should remove it?

Reply:
Maybe it is good enough to just cover this in the existing unit tests...

Reply:
> Do you still think I should remove it?

Yes, please remove it.